runhouse/resources/hardware/utils.py

```diff
@@ -711,3 +760,90 @@ def __str__(self):
         if self.retry:
             s += ", retry in {0}ms".format(self.retry)
         return s
+
+
+###################################
+# KUBERNETES SETUP
+###################################
+def setup_kubernetes(
+    namespace: Optional[str] = None,
+    kube_config_path: Optional[str] = None,
+    context: Optional[str] = None,
+    **kwargs,
+):
+    if kwargs.get("provider") and not kwargs.get("provider") == "kubernetes":
+        raise ValueError(
+            f"Received non kubernetes provider {kwargs.get('provider')} with kubernetes specific "
+            "cluster arguments."
+        )
+
+    if (
+        kwargs.get("server_connection_type")
+        and kwargs.get("server_connection_type") != ServerConnectionType.SSH
+    ):
+        raise ValueError(
+            "Runhouse K8s Cluster server connection type must be set to `ssh`. "
+            f"You passed {kwargs.get('server_connection_type')}."
+        )
+
+    if context and namespace:
+        logger.warning(
+            "You passed both a context and a namespace. Ensure your namespace matches the one in your context.",
+        )
+
+    launcher = kwargs.get("launcher") or configs.launcher
+    if launcher == "local":
+        if context:
+            # check if user passed a user-defined context
+            try:
+                cmd = f"kubectl config use-context {context}"  # set user-defined context as current context
+                subprocess.run(cmd, shell=True, check=True)
+                logger.info(f"Kubernetes context has been set to: {context}")
+            except subprocess.CalledProcessError as e:
+                logger.info(f"Error setting context: {e}")
```

Comment:
wonder if we should actually raise an exception if we fail to set the context or namespace? we don't do it in den either, but this could lead to undesired behavior the user wouldn't want (ex: launching in a different namespace)